### ISSUE TITLE
Invalid receiver on constructor with polymorphic type parameters

### ISSUE THREAD

**Issue by jwaataja** [typetools/checker-framework#2330]
Consider this class
```java
import org.checkerframework.checker.tainting.qual.*;
import java.util.*;

public class A<T extends @Tainted Object> {
    public @Untainted A(@PolyTainted int i) {
    }

    public @Untainted A() {
    }

    public static void f(@PolyTainted int i) {
        new @Untainted A<@PolyTainted Integer>(i);
        new @Untainted A<@PolyTainted Integer>();
    }
}
```

When running the tainting checker, I get this output
```
A.java:13: error: [constructor.invocation.invalid] creation of @Untainted A<@Tainted Integer> <init>() not allowed with given receiver;
        new @Untainted A<@PolyTainted Integer>();
        ^
  found   : @Untainted A<@PolyTainted Integer>
  required: @Untainted A<@Tainted Integer>
```

I don't think there should be an error for this line, it's just calling an empty constructor. This indicates the receiver for a method is invalid, which doesn't make sense for constructors.

Notice, that the line that causes an error is `new @Untainted A<@PolyTainted Integer>()`, not `new @Untainted A<@PolyTainted Integer>(i)`. The error only appears when there are no `@PolyTainted` parameters. For methods with no `@PolyTainted` parameters, the return type would usually default to `@Tainted`, but this shouldn't apply to type parameters. We might want to write `new List<@PolyTainted String>()`, and then add instances of `@PolyTainted String` later.

Regardless, the error message is confusing and worth fixing. We believe the error occurs because the new operator allocates a new `A<@PolyTainted Integer>` and then passes it to the `<init>` method, which expects `@Tainted Integer`. It probably expects `@Tainted Integer` because of the fact there are no `@PolyTainted` parameters.
